src/nwb_conversion_tools/tools/data_transfers.py

```diff
@@ -61,14 +63,143 @@ def get_globus_dataset_content_sizes(
+        You may wish to set this to be longer when transferring very large files.
+        Defaults to 10 minutes.
+
+    Returns
+    -------
+    success : bool
+        If 'display_progress'=False, and the transfer was succesfully initiated, then this function returns True.
+
+        If 'display_progress'=True (the default), then this function returns the total status of all transfers
+        when they either finish or the progress tracking times out.
+    task_ids : list of strings
+        List of the task IDs submitted to globus, if further information is needed to reestablish tracking or terminate.
+    """
+    source_files = [[source_files]] if isinstance(source_files, str) else source_files
+    # assertion check is ensure the logical iteration does not occur over the individual string values
+    assert (
+        isinstance(source_files, list) and source_files and isinstance(source_files[0], list) and source_files[0]
+    ), "'source_files' must be a non-empty nested list-of-lists to indicate batched transfers!"
+    assert destination_folder.is_absolute(), (
+        "The 'destination_folder' must be an absolute path to resolve ambiguity with relative Globus head "
+        "as well as avoiding Globus access permissions to a personal relative path!"
+    )
+
+    folder_content_sizes = dict()
+    task_total_sizes = dict()
+    for j, batched_source_files in enumerate(source_files):
+        paths_file = Path(destination_folder) / f"paths_{j}.txt"
+        # .as_posix() to ensure correct string form Globus expects
+        # ':' replacement for Windows drives
+        source_folder = Path(batched_source_files[0]).parent.as_posix().replace(":", "")
+        destination_folder_name = Path(destination_folder).as_posix().replace(":", "")
+        with open(file=paths_file, mode="w") as f:
+            for source_file in batched_source_files:
+                file_name = Path(source_file).name
+                f.write(f"{file_name} {file_name}\n")
+
+        transfer_command = (
+            f"globus transfer {source_id}:{source_folder} {destination_id}:{destination_folder_name} "
+            f"--batch {paths_file}"
+        )
+        transfer_message = deploy_process(
+            command=transfer_command,
+            catch_output=True,
+        )
+        task_id = re.findall(
+            pattern=(
+                "^Message: The transfer has been accepted and a task has been created and queued for "
+                "execution\nTask ID: (.+)$"
+            ),
+            string=transfer_message,
+        )
+        paths_file.unlink()
+        assert task_id, f"Transfer submission failed! Globus output:\n{transfer_message}."
+
+        if source_folder not in folder_content_sizes:
+            contents = get_globus_dataset_content_sizes(globus_endpoint_id=source_id, path=source_folder)
+            folder_content_sizes.update({source_folder: contents})
+        task_total_sizes.update(
```

Comment:
It seems a little bit complicated to see what structure you get at the end. Flagging it so I can think a little bit better about it. Maybe there is a better way of writing this.

Reply:
double-nested dict tree - outer level is unique common parent folders, inner level is filename in that folder with value equal to size in bytes.

Not that the JSON content API request is that expensive, but still, you'd really only want it to have to grab the info once per unique folder.